moabb/contexts/base.py

```diff
@@ -44,7 +46,29 @@ def __init__(self, datasets, pipelines):
                 raise(ValueError("pipelines must only contains Pipelines instance"))
         self.pipelines = pipelines
 
-
+    @abstractmethod
+    def score(self, clf, X, y, groups, n_jobs=1):
+        '''
+        Return score 
+        '''
+        pass
+
+    @abstractmethod
+    def prepare_data(self, dataset, subjectlist):
+        '''
+        Given dataset, fetch data from subjects
+
+        Parameters:
+            dataset:       Dataset instance
+            subjectlist:   List of ids **(strings? numbers?)** for subjects
+
+        Output:
+            X:       ndarray (trials, channels, timepoints) of data
+            y:       ndarray (trials,) **1 or 2d** of labels
+            groups:  ndarray (trials,) specifying which subject each trial belongs to
```

Comment:
No, this is not a bug.
For example, with your PR you reverted the fact that for many dataset i used session for group index in order to do a leave-one session out CV.

we can not enforce group to be subjects, it could be anything that is relevant for the CV.